### PR TITLE
DialV2: add support for custom timeouts

### PR DESCRIPTION
**Pull request by fialhopm**
Add support for custom transport timeouts. Discussed [here](https://github.com/gebn/bmc/issues/62#issuecomment-1878479915).

### CODE REVIEW

**Comment by gebn** [comment]
What do you think about using the [functional options pattern][] here? It would result in an API like `bmc.DialV2(addr, bmc.WithTimeout(timeout))`. Those not wanting to specify a timeout would omit the second arg. Admittedly it would be verbose for this, but more extensible for the future.

[functional options pattern]: https://www.yellowduck.be/posts/the-functional-options-pattern-in-go

File: bmc.go [hunk 1]
```diff
@@ -23,36 +23,55 @@ var (
 	namespace = "bmc"
 )
 
+type Config struct {
```
Comment (fialhopm) [inline bmc.go:26]: Should this have a more descriptive name? E.g., `V2SessionlessTransportConfig` or `DialConfig`.

Comment (gebn) [reply bmc.go:26]: I agree Config is a little too generic. I would use DialConfig and DialConfigOption. I'm not sure what the conventions are around exporting DialConfig - can we get away with dialConfig?

Comment (fialhopm) [reply bmc.go]: > can we get away with dialConfig?

I believe we can: https://pkg.go.dev/golang.org/x/net/idna#Option

**Pull request merged**